_pytest/python_api.py

```diff
@@ -450,6 +450,9 @@ def approx(expected, rel=None, abs=None, nan_ok=False):
     elif isinstance(expected, Mapping):
         cls = ApproxMapping
     elif isinstance(expected, Sequence) and not isinstance(expected, String):
+        for value in expected:
+            if isinstance(value, (Sequence, String, Mapping)):
```

Comment:
I believe it is better to do check for numbers, and provide a better message to locate the problem:

```python
        number_types = six.integer_types + (float,)
        for index, value in enumerate(expected):
            if not isinstance(value, number_types):
				msg = 'Sequence must contain only numbers, but found {!r} ({!r}) at index {}'
                raise TypeError(msg.format(value, type(value), index))
```

Reply:
We also should add a test for this 😉

Reply:
I cant just check for int, float, and long though. I would also have to check for each numpy type too.

Reply:
@nicoddemus How should I approach numpy types?

Reply:
There's already a `elif _is_numpy_array(expected):` check near the start of the if/elif chain, so `numpy.array` are covered already.

There's the situation where the user passes a builtin `list` containing `numpy` numbers, but that seems rare to me. So we can either:

1. Also check every possible number type, including numpy's:

   ```python
   # _get_numpy_types() returns a tuple (numpy.int32, numpy.int64, ...) if numpy is available,
   # otherwise an empty tuple
   number_types = six.integer_types + (float,) + self._get_numpy_types()  
   ```

2. Consider this overkill and just fail when we encounter a `list` containing numpy types. In this case
   should make sure to show the data type in the error message (like in my previous example).


I think it is worthwhile trying to go with 1), this will make things more convenient for our users at the cost of a little more
code.   

What do you folks think?